client/js/_author/components/assessments/preview_container.jsx

```diff
@@ -7,32 +7,30 @@ export default class PreviewContainer extends React.Component {
     getAssessmentOffered: React.PropTypes.func.isRequired,
     assessmentPlayerUrl: React.PropTypes.string.isRequired,
     apiUrl: React.PropTypes.string.isRequired,
-    authoringToolPreviewSettings: React.PropTypes.arrayOf(React.PropTypes.string).isRequired,
   };
 
   static hasOffered(assessment) {
     return !(_.isUndefined(assessment.assessmentOffered)
       || _.isEmpty(assessment.assessmentOffered[0]));
   }
 
+  componentDidMount() {
+    const assessment = this.props.assessment;
+    if (!PreviewContainer.hasOffered(assessment)) {
+      this.props.getAssessmentOffered(assessment.bankId, assessment.id);
+    }
+  }
+
   buildEmbedUrl() {
-    const { assessmentPlayerUrl, apiUrl, assessment } = this.props;
+    const { assessmentPlayerUrl, apiUrl, assessment, unlockNext } = this.props;
 
     const bankId = assessment.bankId;
     const assessmentOfferedId = _.get(assessment, 'assessmentOffered[0].id');
-    const previewSettings = _.get(this.props, 'authoringToolPreviewSettings', []).join('&');
+    const previewSettings = [`unlock_next=${unlockNext}`].join('&');
```

Comment:
I was thinking of leaving that, so that adding more settings to it would be real easy.